src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -811,44 +811,22 @@ internal void SendModelEvent(Guid modelGuid, string eventName)
             HasUnsavedChanges = true;
         }
 
-        internal void UpdateModelValue(Guid modelGuid, string propertyName, string value)
+        internal void UpdateModelValue(IEnumerable<Guid> modelGuids, string propertyName, string value)
         {
-            var retrievedModel = GetModelInternal(modelGuid);
-            if (retrievedModel == null)
+            if (modelGuids == null || (!modelGuids.Any()))
+                throw new ArgumentNullException("modelGuids");
+
+            var retrievedModels = GetModelsInternal(modelGuids);
+            if (!retrievedModels.Any())
                 throw new InvalidOperationException("UpdateModelValue: Model not found");
 
-            var handled = false;
-            var nodeModel = retrievedModel as NodeModel;
-            if (nodeModel != null)
+            using (new ModelModificationUndoHelper(undoRecorder, retrievedModels))
             {
-                using (new NodeModificationUndoHelper(undoRecorder, nodeModel))
+                foreach (var retrievedModel in retrievedModels)
                 {
-                    handled = nodeModel.UpdateValue(propertyName, value, undoRecorder);
+                    retrievedModel.UpdateValue(propertyName, value, undoRecorder);
                 }
             }
-            else
-            {
-                // Perform generic undo recording for models other than node.
-                RecordModelForModification(retrievedModel, UndoRecorder);
-                handled = retrievedModel.UpdateValue(propertyName, value, undoRecorder);
-            }
-
-            if (!handled) // Method call was not handled by any derived class.
-            {
-                string type = retrievedModel.GetType().FullName;
-                string message = string.Format(
-                    "ModelBase.UpdateValue call not handled.\n\n" +
-                    "Model type: {0}\n" +
-                    "Model GUID: {1}\n" +
-                    "Property name: {2}\n" +
-                    "Property value: {3}",
-                    type, modelGuid, propertyName, value);
-
-                // All 'UpdateValue' calls must be handled by one of the 
-                // ModelBase derived classes that the 'UpdateModelValue'
-                // is intended for.
-                throw new InvalidOperationException(message);
-            }
```

Comment:
We no longer throw an exception if a subset of the retrieved `ModelBase` object cannot handle the `UpdateValueCore` call. For example, selecting a `NodeModel` and `NoteModel` then perform a set-lacing command, `NoteModel` will not be able to handle the `UpdateValueCore` method call.